docs/pages/ESSupport.md

```diff
@@ -0,0 +1,368 @@
+        const name = "Component";
+        return TemplateAssembler.getTemplateComponent(getMethods,
+		`sap.fe.templates.Page.${name}`, {
+			metadata: {
+				properties: {
+					"templateName": {
+						"type": "string",
+						"defaultValue": "sap.fe.templates.Page.view.Page"
+					}
+				},
+				"manifest": "json"
+			}
+        }
+)});
+```
+
+## Library
+For libraries some special cases require following best practices.
+
+### sap/ui/core/Core#initLibrary: Usage of Object Expression, Spread Element and Template Litaral
+Typically in the `library.js` of your library the library is initialized. The object which is given to the `sap/ui/core/Core#initLibrary` call musst not include any `Object Expression`, `Spread Element` or `Template Literal` with expression.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.getCore().initLibrary({
+    name: "my.lib"
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+const mylib = {
+    key: "name"
+};
+sap.ui.getCore().initLibrary({
+    [mylib.key] : "my.lib",
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+const mylib = {
+    name: "my.lib"
+};
+sap.ui.getCore().initLibrary({
+    ...mylib
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+const name = "my"
+sap.ui.getCore().initLibrary({
+    name: `{$name}.lib`
+```
+
+### JSDoc
+For libraries the JSDoc output might be relevant. Therefore, please follow the below best practices.
+
+#### sap.ui.define/sap.ui.require: with Arrow function expression
+The JSDoc block of the module has to be written above the arrow function and not above the `sap.ui.define/sap.ui.require` command.
+
+**✅ Best Practice:**
+
+```javascript
+sap.ui.define(["Bar"],
+    /**
+     * JSDoc block here
+     */
+    (Bar) => Bar.extends("Foo", {...})
+);
+```
+
+**⛔️ Not supported:**
+```javascript
+/**
+ * JSDoc block here (Not detected properly!)
+ */
+sap.ui.define(["Bar"], (Bar) => Bar.extends("Foo", {...}));
+```
+
+#### sap.ui.define/sap.ui.require: Returning a class definition
+The class declaration should not be returned directly. Declare the class and return the class in a separate statement. JSDoc treats the the class declaration as a return statement and does not recognize any JSDoc if such is provided right above the class declaration.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.define(["Bar"], function(Bar){
+    /**
+     * JSDoc block here
+     */
+    class Foo extends Bar {
+        make () {
+        ...
+        }
+    }
+
+    return Foo;
+});
+```
+
+**⛔️ Not supported:**
+```javascript
+sap.ui.define(["Bar"], function(Bar){
+    /**
+     * JSDoc block here
+     */
+    return class Foo extends Bar {
+        make () {
+        ...
+        }
+    }
+});
+```
+
+
+#### Multiple variable declarations: Syntax.ObjectPattern, Syntax.VariableDeclaration, Syntax.ArrowFunctionExpression
+
+**⛔️ Not supported:**
+```javascript
+({a} /* Syntax.ObjectPattern */, b) => { // Syntax.ArrowFunctionExpression
```

Comment:
when using ES6, I would expect the usage of `var` will be avoided. With that declaring a variable with the same name is obsolete. IMHO I would not document this restriction because its such a bad practice that writing it down is worse :-)

Reply:
Fine for me